### PR TITLE
AP_DDS: Added Micro-XRCE-DDS-GEN instal to Dockerfile.

### PR DESCRIPTION
**Pull request by PQuill33**
# Associated Issue #
[#28348](https://github.com/ArduPilot/ardupilot/issues/28348)

# What Changed #
Update to Ardupilot Dockerfile used to build ardupilot code base with --enable-dds flag. The update is an installation of the Micro-XRCE-DDS-Gen dependency to use the enable-dds flag.

# How to run #
* Browse to the `ardupilot` source code folder.
* Build the docker developer image **the first time**:
```bash
docker build . -t ardupilot --build-arg USER_UID=$(id -u) --build-arg USER_GID=$(id -g)
```
* Run the developer container with:
```bash
docker run --rm -it -v "$(pwd):/ardupilot" -u "$(id -u):$(id -g)" ardupilot:latest bash
```
* The container will mount the ardupilot source code as volume. Now configure the target board:
```bash
./waf configure --board <board-name> --enable-dds
```
where `<board-name>` must be one of the supported boards, like `pixracer`.
* Once Configuration is successful build the code with:
```bash
./waf <type>
```
where `<type>` is one of the supported vehicle types, like `copter` or `plane`.

**Note** It might be convenient to delete the `build` folder before building new software.

### CODE REVIEW

**Comment by Ryanf55** [comment]
This conflicts with the established workflow here:
https://ardupilot.org/dev/docs/ros2.html#installation-docker

Consider testing that.

**Review by Ryanf55 — APPROVED** [review]

**Comment by PQuill33** [comment]
> This conflicts with the established workflow here: https://ardupilot.org/dev/docs/ros2.html#installation-docker
> 
> Consider testing that.
> 
> That said, this doesn't hurt, so we could merge it.

I tested the established workflow mentioned, and mounted it to my AP repo and got build errors, not sure if there's something missing there or not

Screenshot from 2024-10-08 11-47-38

**Comment by Ryanf55** [comment]
> > This conflicts with the established workflow here: https://ardupilot.org/dev/docs/ros2.html#installation-docker
> > Consider testing that.
> > That said, this doesn't hurt, so we could merge it.
> 
> I tested the established workflow mentioned, and mounted it to my AP repo and got build errors, not sure if there's something missing there or not
> 
> Screenshot from 2024-10-08 11-47-38

image

Try building in verbose mode.

File: Dockerfile
Comment (khancyr) [inline Dockerfile:52]: add to .ardupilot_env

Comment (PQuill33) [reply Dockerfile:52]: Done, I updated to export PATH in .ardupilo_env in latest commit.

**Review by khancyr — CHANGES_REQUESTED** [review]
Need rebase to get rid of merge commit and merge the two commit into one. And it is good to go !

**Comment by Ryanf55** [comment]
> Need rebase to get rid of merge commit and merge the two commit into one. And it is good to go !

Done. I force-pushed it.

**Review by khancyr — APPROVED** [review]
LGTM

**Comment by PQuill33** [comment]
Awesome, thanks guys! Looks like it's ready to merge then, should I close the PR or wait until it is merged?

**Comment by Ryanf55** [comment]
Please leave it open. A maintainer will merge it shortly.

**Pull request merged**
